uber/site_sections/preregistration.py

```diff
@@ -198,12 +198,15 @@ def prereg_payment(self, session, payment_id, stripeToken):
         else:
             message = charge.charge_cc(stripeToken)
 
-        if message:
+        if message and charge.total_cost:
             raise HTTPRedirect('index?message={}', message)
 
         for attendee in charge.attendees:
             attendee.paid = c.HAS_PAID
             attendee.amount_paid = attendee.total_cost
+            if attendee.promo_code_id:
```

Comment:
You'll wanna change this to `if attendee.promo_code` due to the changes you need to make to the relationship. You also _shouuuldn't_ need the next line since SQLAlchemy shouuuuuld magically do that bit for you.